node/src/accountant/scanners/mod.rs

```diff
@@ -854,24 +866,31 @@ impl Scanner<RetrieveTransactions, ReceivedPayments> for ReceivableScanner {
     }
 
     fn finish_scan(&mut self, msg: ReceivedPayments, logger: &Logger) -> Option<NodeToUiMessage> {
-        if msg.payments.is_empty() {
-            info!(
-                logger,
-                "No newly received payments were detected during the scanning process."
-            );
-
-            match self
-                .persistent_configuration
-                .set_start_block(msg.new_start_block)
-            {
-                Ok(()) => debug!(logger, "Start block updated to {}", msg.new_start_block),
-                Err(e) => panic!(
-                    "Attempt to set new start block to {} failed due to: {:?}",
-                    msg.new_start_block, e
-                ),
+        match msg.scan_result {
+            Ok(payments_and_start_block) => {
+                self.handle_new_received_payments(&payments_and_start_block, msg.timestamp, logger);
             }
-        } else {
-            self.handle_new_received_payments(&msg, logger)
+            Err(e) => match e {
+                ReceivedPaymentsError::ExceededBlockScanLimit(max_block_count) => {
+                    debug!(logger, "Writing max_block_count({})", max_block_count);
+                    self.persistent_configuration
+                        .set_max_block_count(Some(max_block_count))
+                        .map_or_else(|_| {
```

Comment:
Despite this piece of code will very likely have to be erased, map_or_else is believed to be hard to read, at least such voices have been heard in our team (including from me), so it's recommend to replace it by a sequence of two other methods like 

something_happening_before + map + map_err

Those two separate maps of different kind read better.